File: JAVA/media.java
import java.util.Scanner;

public class media {
    public static void main(String args[]){
        Scanner scanner = new Scanner(System.in);
        int[] nota = new int[4];
        int[] peso = new int[4];

        for (int i = 1; i < 4; i++){
            System.out.println("Qual a nota" + i + "?");
            nota[i] = scanner.nextInt();
            
            System.out.println("Qual o peso" + i + "?");
            peso[i] = scanner.nextInt();
        }
        
        int pesofinal = peso[1] + peso[2] + peso[3];
        int notafinal = (nota[1] * peso[1]) + (nota[2] * peso[2]) + (nota[3] * peso[3]);
        double resultado = notafinal / pesofinal;

        System.out.println(resultado);
    }
}

File: C/media.c
#include <stdio.h>

// Solicita a operação
int main(){
    int idade;
    int nota[3];
    int peso[3];

    for (int i = 1; i < 4; i++) {

    printf("Qual a nota %d?\n", i);
    scanf("%d", &nota[i]);

    printf("Qual o peso %d?\n", i);
    scanf("%d", &peso[i]);

    }
    int pesofinal = peso[1] + peso[2] + peso[3];
    int notafinal = (nota[1] * peso[1]) + (nota[2] * peso[2]) + (nota[3] * peso[3]);
    printf("A media ponderada e: %.2f\n", (float)(notafinal / pesofinal));

    float num1;
    printf("Digite qualquer coisa para encerrar o programa. \n");
    scanf("%s", &num1);
    return 0;
}